client/src/app/components/stardust/block/payload/TransactionPayload.tsx

```diff
@@ -1,104 +1,181 @@
-import React, { ReactNode } from "react";
-import { TransactionPayloadProps } from "./TransactionPayloadProps";
-import { TransactionPayloadState } from "./TransactionPayloadState";
+import {
+    AddressUnlockCondition,
+    CommonOutput,
+    ExpirationUnlockCondition,
+    GovernorAddressUnlockCondition,
+    ReferenceUnlock,
+    SignatureUnlock,
+    StateControllerAddressUnlockCondition,
+    UnlockConditionType,
+    UnlockType,
+    Utils,
+} from "@iota/sdk-wasm/web";
+import React, { useContext, useEffect, useState } from "react";
+import NetworkContext from "~/app/context/NetworkContext";
+import { Bech32AddressHelper } from "~/helpers/stardust/bech32AddressHelper";
 import transactionPayloadMessage from "~assets/modals/stardust/block/transaction-payload.json";
-import NetworkContext from "../../../../context/NetworkContext";
-import AsyncComponent from "../../../AsyncComponent";
 import Modal from "../../../Modal";
 import Input from "../../Input";
 import Output from "../../Output";
 import Unlocks from "../../Unlocks";
+import { IPreExpandedConfig } from "../../interfaces";
 import "./TransactionPayload.scss";
+import { TransactionPayloadProps } from "./TransactionPayloadProps";
 
 /**
  * Component which will display a transaction payload.
  */
-class TransactionPayload extends AsyncComponent<TransactionPayloadProps, TransactionPayloadState> {
-    /**
-     * The component context type.
-     */
-    public static contextType = NetworkContext;
+const TransactionPayload: React.FC<TransactionPayloadProps> = ({ network, inputs, unlocks, outputs, header, isLinksDisabled }) => {
+    const [inputsPreExpandedConfig, setInputsPreExpandedConfig] = useState<IPreExpandedConfig[]>([]);
+    const { bech32Hrp } = useContext(NetworkContext);
 
-    /**
-     * The component context.
-     */
-    public declare context: React.ContextType<typeof NetworkContext>;
+    const OUTPUT_EXPAND_CONDITIONS: UnlockConditionType[] = [
+        UnlockConditionType.Address,
+        UnlockConditionType.StateControllerAddress,
+        UnlockConditionType.GovernorAddress,
+    ];
 
-    /**
-     * Create a new instance of TransactionPayload.
-     * @param props The props.
-     */
-    constructor(props: TransactionPayloadProps) {
-        super(props);
+    const INPUT_EXPAND_CONDITIONS: UnlockConditionType[] = [...OUTPUT_EXPAND_CONDITIONS, UnlockConditionType.Expiration];
 
-        this.state = {
-            isFormattedBalance: true,
-        };
-    }
+    useEffect(() => {
+        if (bech32Hrp) {
+            // given the unlocks, expand the correct address unlock condition
+            const inputsPreExpandedConfig: IPreExpandedConfig[] = inputs.map((input, idx) => {
```

Comment:
Should we move this bad boy out somewhere ? 🤔

Reply:
moved the code around